src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;
+	private String errorMsg;
+	private boolean userCanceled;

```

Comment:
Seems you are right. Without this "race condition" constraint due to the Swing progress bar thread (Cf. BaselineCorrectorSetupDialog), that is to say, in every other cases, this additional variable seems useless. Let's remove it...